### PR TITLE
Shortcuts modal

### PR DESCRIPTION
**Pull request by samuel-peers**
Associated Issue: #4122

### Summary of Changes

Description of actions and keyboard shortcuts

### Screenshots
untitled

### CODE REVIEW

File: assets/panel/debugger.properties [hunk 1]
```diff
@@ -152,6 +152,30 @@ projectTextSearch.key=CmdOrCtrl+Shift+F
 # modal for searching functions in a file.
 functionSearch.key=CmdOrCtrl+Shift+O
 
+# LOCALIZATION NOTE (functionSearch.key): A key shortcut to toggle
```
Comment (flodolo) [inline assets/panel/debugger.properties]: Right comment, wrong reference ID (they're all wrong in this section).

File: assets/panel/debugger.properties [hunk 2]
```diff
@@ -702,3 +726,43 @@ ctrl=Ctrl
 # LOCALIZATION NOTE (anonymous): The text that is displayed when the
 # display name is null.
 anonymous=(anonymous)
+
+# LOCALIZATION NOTE (shortcuts.toggleBreakpoint): text describing
+# keyboard shortcut action for toggling breakpoint
+shortcuts.toggleBreakpoint=Toggle Breakpoint
+
+# LOCALIZATION NOTE (shortcuts.toggleBreakpoint): text describing
```
Comment (flodolo) [inline assets/panel/debugger.properties]: Same issue

File: assets/panel/debugger.properties [hunk 3]
```diff
@@ -152,6 +152,30 @@ projectTextSearch.key=CmdOrCtrl+Shift+F
 # modal for searching functions in a file.
 functionSearch.key=CmdOrCtrl+Shift+O
 
+# LOCALIZATION NOTE (functionSearch.key): A key shortcut to toggle
+# breakpoints.
+toggleBreakpoint.key=CmdOrCtrl+B
+
+# LOCALIZATION NOTE (functionSearch.key): A key shortcut to toggle
+# the conditional breakpoint panel.
+toggleCondPanel.key=CmdOrCtrl+B
```
Comment (jasonLaster) [inline assets/panel/debugger.properties]: i think we should have all of these keys already defined

File: assets/panel/debugger.properties [hunk 4]
```diff
@@ -152,6 +152,30 @@ projectTextSearch.key=CmdOrCtrl+Shift+F
...
+
+# LOCALIZATION NOTE (functionSearch.key): A key shortcut to toggle
+# pause and resume.
+pauseOrResume.key=F8
```
Comment (jasonLaster) [inline assets/panel/debugger.properties]: We probably don't need F8 in properties, we can just include the string directly

File: src/utils/prefs.js [hunk 5]
```diff
@@ -27,8 +27,8 @@ if (isDevelopment()) {
   pref("devtools.debugger.features.project-text-search", true);
   pref("devtools.debugger.features.async-stepping", true);
   pref("devtools.debugger.features.wasm", true);
-  pref("devtools.debugger.features.shortcuts", false);
-  pref("devtools.debugger.features.utilsBar", false);
+  pref("devtools.debugger.features.shortcuts", true);
+  pref("devtools.debugger.features.utilsBar", true);
```
Comment (jasonLaster) [inline src/utils/prefs.js]: oh oops, lets get rid of utilsBar and use shortcuts

File: src/utils/prefs.js [hunk 6]
```diff
@@ -54,8 +54,8 @@ export const features = new PrefsHelper("devtools.debugger.features", {
   asyncStepping: ["Bool", "async-stepping", false],
   projectTextSearch: ["Bool", "project-text-search", true],
   wasm: ["Bool", "wasm", true],
-  shortcuts: ["Bool", "shortcuts", false],
-  utilsBar: ["Bool", "utilsBar", false]
+  shortcuts: ["Bool", "shortcuts", true],
```
Comment (jasonLaster) [inline src/utils/prefs.js]: it should default to false

File: src/components/shared/Modal.js [hunk 7]
```diff
@@ -21,6 +22,10 @@ export class Modal extends Component {
     self.onClick = this.onClick.bind(this);
   }
 
+  overrideHeight() {
+    return { height: this.props.height };
+  }
```
Comment (jasonLaster) [inline src/components/shared/Modal.js]: can we do this in CSS so the respective css modals would set the height?

File: src/components/shared/Modal.js [hunk 8]
```diff
@@ -30,7 +35,11 @@ export class Modal extends Component {
 
     return (
       <div className="modal-wrapper" onClick={this.props.handleClose}>
-        <div className={classnames("modal", status)} onClick={this.onClick}>
+        <div
```
Comment (jasonLaster) [inline src/components/shared/Modal.js]: perhaps we want to be able to pass in a modal class name that can override the height? `className={classnames("modal", modalClass)}`

File: src/components/ShortcutsModal.js [hunk 9]
```diff
@@ -10,8 +11,80 @@ export class ShortcutsModal extends Component {
...
+          formatKeyShortcut(L10N.getStr("functionSearch.key"))
+        )}
+      </ul>
+    );
+  }
+
+  renderShortcutsContent() {
+    return (
+      <div className="shortcuts-content">
+        <h2>Editor</h2>
```
Comment (jasonLaster) [inline src/components/ShortcutsModal.js]: lets L10N these as well

File: assets/panel/debugger.properties [hunk 10]
```diff
@@ -164,6 +164,18 @@ toggleCondPanel.key=CmdOrCtrl+Shift+B
 # step out.
 stepOut.key=Shift+F11
 
+# LOCALIZATION NOTE (stepOut.key): Sections header in
+# the shortcuts modal for keyboard shortcuts related to editing.
+shortcuts.header.editor=Editor
+
+# LOCALIZATION NOTE (stepOut.key): Sections header in
```
Comment (flodolo) [inline assets/panel/debugger.properties]: And, again, wrong string ID all over the place

**Pull request merged**